components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Protocol/src/Services/SubscriptionServices.cs

```diff
@@ -1118,6 +1124,35 @@ private class MonitoredItemWrapper {
                             eventFilter.AddSelectClause(ObjectTypeIds.ConditionType, "Retain");
                         }
                     }
+
+                    var sb = new StringBuilder();
+
+                    // let's loop thru the select clause and setup the field names
+                    foreach (var selectClause in eventFilter.SelectClauses) {
+                        sb.Clear();
+                        for (var i = 0; i < selectClause.BrowsePath.Count; i++) {
+                            if (i == 0) {
+                                if (selectClause.BrowsePath[i].NamespaceIndex != 0) {
+                                    sb.Append(session.NamespaceUris.GetString(selectClause.BrowsePath[i].NamespaceIndex));
+                                    sb.Append("#");
+                                }
+                            }
+                            else {
+                                sb.Append("/");
+                            }
+                            sb.Append(selectClause.BrowsePath[i].Name);
+                        }
+
+                        if (sb.Length == 0) {
```

Comment:
Guess you could merge these statements. And remove the empty line below.

Reply:
This I don't understand. Elaborate

Reply:
Sorry, the comment should be one line below. There is a nested if statement in this. I think these two if statements can be merged.

Reply:
No, not in my opinion. I think this is cleaner expression of intent

Reply:
All for that!